### DISCUSSION TITLE
Introduce async pattern in Prism WPF application

### DISCUSSION THREAD

**Discussion by bauradar** [PrismLibrary/Prism#2967] [WPF]
Hello,

I am using Prism for a number of years and it is a very helpful framework.

In order to work with a REST api as datasource (instead of a direct client - server - app) I need to go async. By using the extension to the Task I can build fire&forget scenarios and also use a callback.

But: What is the best way to start async methods from the Prism bootstrapper? I cannot use a callback and come back to the next bootsrapper step. The async loaded data needs to be in place before the shell will be created.

Any experience on this subject is very welcome.

Thanks 

Niko

**Comment by bennoremec** [comment]
1. Rethink your approach, wheter you really can't show anything until the data is loaded. This is not very user friendly
2. If it's not possible, then look at solutions like this: https://stackoverflow.com/a/70934542/66207
